### PR TITLE
Adding upgrade tests across image types

### PR DESCRIPTION
**Pull request by bdbauer**
@maisem

---
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/test-infra/567)

### CODE REVIEW

**Comment by maisem** [comment]
cc: @vishh @pwittrock @spxtr 

kubernetes/kubernetes#32588 needs to be merged (and cherry-picked to 1.2, 1.3 and 1.4) before this can go in.

**Comment by maisem** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml, line 646 at r1](https://reviewable.io:443/reviews/kubernetes/test-infra/567#-KR_IZjmUt8LhowPFtSC:-KR_IZjmUt8LhowPFtSD:b-v2hb9o) ([raw file](https://github.com/kubernetes/test-infra/blob/e364be4f8cd23fdb06bf0aa84a6e8be317880697/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml#L646)):*
> ```YAML
>                 export JENKINS_USE_SKEW_TESTS="true"
>                 export KUBE_GKE_IMAGE_TYPE="{image-old}"
>                 export PROJECT="kube-gke-up-{version-infix}-up-clu-n"
> ```

s/kube-gke/gke

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/567)*

**Comment by maisem** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/kubernetes/test-infra/567#-:-KR_bjqK2IZo5T0ReS2M:bnfp4nl)*

**Comment by foxish** [comment]
@maisem @bdbauer https://github.com/kubernetes/kubernetes/pull/32614 and https://github.com/kubernetes/kubernetes/pull/32615 are now merged into release-1.4 and release-1.3 respectively. Can we go ahead with this change now? The cherrypick on 1.2 will need some more work as there are merge conflicts.

**Comment by maisem** [comment]
@foxish - I still need to create the jenkins projects for these, will do that shortly.

File: jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml [hunk 1]
```diff
@@ -541,9 +541,10 @@
             job-env: |
                 export E2E_OPT="--check_version_skew=false"
                 export E2E_UPGRADE_TEST="true"
-                export GINKGO_UPGRADE_TEST_ARGS="--ginkgo.focus=\[Feature:MasterUpgrade\] --upgrade-target=ci/latest-{version-new}"
+                export GINKGO_UPGRADE_TEST_ARGS="--ginkgo.focus=\[Feature:MasterUpgrade\] --upgrade-target=ci/latest-{version-new} --upgrade-image=container_vm"
```
Comment (maisem) [inline jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e-gke.yaml]: Just to be safe here, remove it from the old jobs, we plan on removing these jobs in a couple of weeks anyways.
i.e. "--upgrade-image=container_vm"

**Comment by maisem** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](http://reviewable.kubernetes.io:80/reviews/kubernetes/test-infra/567#-:-KRejJeLgvqArF27prH5:bnfp4nl)*

**Pull request merged**